lsp-dart-flutter-daemon.el

```diff
@@ -93,26 +105,38 @@ PARAMS is the optional method params."
                 ("device.added" (lsp-dart-flutter-daemon--device-added params))
 
                 ("daemon.logMessage" (lsp-dart-flutter-daemon--log level? message?)))
-            (let* ((command (alist-get id lsp-dart-flutter-daemon-commands))
-                   (callback (plist-get command :callback)))
-              (when command
-                (setq lsp-dart-flutter-daemon-commands
-                      (lsp-dart-remove-from-alist id lsp-dart-flutter-daemon-commands)))
-              (when callback
-                (when result?
-                  (funcall callback result?))))))
+            (message "%S" `(,id ,result? ,event?))
+            (when callback
+              (when result?
+                (funcall callback result?)))))
         (lsp-dart-flutter-daemon--raw->response raw-response)))
 
+(defun lsp-dart-flutter-daemon--process-filter (process message &optional callback)
+  "Handle MESSAGE from PROCESS."
+  (message "[filter] Got %S from %S" message process)
+  (lsp-dart-flutter-daemon--handle-responses message callback)
+  (lambda ()
+    (delete-process process)))
+
 (defun lsp-dart-flutter-daemon--send (method &optional params callback)
   "Build and send command with METHOD with optional PARAMS.
 Call CALLBACK if provided when the receive a response with the built id
 of this command."
   (unless (lsp-dart-flutter-daemon--running-p)
     (lsp-dart-flutter-daemon-start))
   (let* ((id (lsp-dart-flutter-daemon--generate-command-id))
-         (command (lsp-dart-flutter-daemon--build-command id method params)))
-    (add-to-list 'lsp-dart-flutter-daemon-commands (cons id (list :callback callback)))
-    (comint-send-string (get-buffer-process lsp-dart-flutter-daemon-buffer-name) command)))
+         (command (lsp-dart-flutter-daemon--build-command id method params))
+         (client (open-network-stream
```

Comment:
I think we should handle connection errors similar to how `dap--open-network-stream` does, right?